### ISSUE TITLE
Newsletter Verification Mail & Confirmation email when creating a new customer are not sent

### ISSUE THREAD

**Issue by MarcoConv** [PrestaShop/PrestaShop#17097]
Prestashop is not sending any verification mail to the customer after the registration. He is automatic enabled for newsletters.
> Edit by Khouloud

Steps to reproduce the issue:
1. Configure the ps_emailsubscription module
2. Enable the options: "Would you like to send a verification email after subscription?" & "Would you like to send a confirmation email after subscription?"
3. Save
4. Go to the FO => Create a new customer with selecting this option "Sign up for our newsletter"
5. Check that emails are not sent (only the welcome mail)
6. Same issue from creating a new customer from the checkout page

**Comment by micka-fdz** [comment]
Hello @MarcoConv, you can activate email verification/confirmation in module configuration.

**Comment by MarcoConv** [comment]
It is activated, but it is not working. After the newsletter registration the value is directly set to 1 without any verification mail.

**Comment by khouloudbelguith** [comment]
Hi @MarcoConv,

I did not manage to reproduce the issue with PS1.7.6.2 & the ps_emailsubscription module v2.5.1.
I attached a screen record
https://drive.google.com/file/d/1ExJdZwO9qoHfsKh6QxZV4g6_oeJYLhPi/view
Thanks to check & feedback.

**Comment by MarcoConv** [comment]
Hi, 
Thanks for your feedback.

It is not about the registration in this field. I mean the checkbox at the customer registration form.

![image](https://user-images.githubusercontent.com/59735886/72872639-217dd880-3cee-11ea-9631-e71714f14077.png)

**Comment by khouloudbelguith** [comment]
@MarcoConv, thanks for your feedback.
I manage to reproduce the issue with PS1.7.6.3 & PS1.7.52 and  ps_emailsubscription v2.5.1 & v2.3.0.
In fact, when creating a new customer from the "Create an account" or Checkout page with selecting this option "Sign up for our newsletter", emails "verification email" & "confirmation email" are not sent.
They are sent only when subscribing from the `submitNewsletter` block as reported here: https://github.com/PrestaShop/PrestaShop/issues/9810
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by fabian-lange** [comment]
Where can I look into the Roadmap?

Far more than a year has passed by now, and this Issue still remains (PS-Verions 1.7.7.5, Module-Version 2.7.0).
Just want to mention that this Bug is making newsletter-subscription on registration illegal in a large part of europe.

When is a fix to be expexted?

**Comment by matks** [comment]
> Where can I look into the Roadmap?

https://github.com/PrestaShop/PrestaShop/projects

> Far more than a year has passed by now, and this Issue still remains (PS-Verions 1.7.7.5, Module-Version 2.7.0). Just want to mention that this Bug is making newsletter-subscription on registration illegal in a large part of europe.
> 
> When is a fix to be expected?

If you find that this issue is critical for you, and it's important to have it fixed soon, then you can _invest_ into having it done. You can hire a developer to fix it, or if you are a developer yourself, you can try doing it on your own. PrestaShop is an open source, community project -- that's what the open source strength is all about.

**Comment by florine2623** [comment]
Indeed, only the Welcome email is sent :
<img width="879" alt="Screenshot 2023-10-12 at 14 53 20" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/4500867a-277e-4c9a-82f4-345aaa91524f">

It should also sent an email for the newsletter confirmation email :
<img width="856" alt="Screenshot 2023-10-12 at 14 54 21" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/d654c79e-5a77-4275-96aa-2ab9c16c6c6f">

And the newsletter verif email : 
<img width="875" alt="Screenshot 2023-10-12 at 14 54 30" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/2c7b0f5e-f094-4319-8016-9a3dd43f1a51">

**Comment by micalm** [comment]
This issue is still happening on PrestaShop 8.2.1 with `ps_emailsubscription` 2.8.3.

**Comment by paulnoelcholot** [comment]
Hello @micalm 

Thank you for your feedback, could you please let us know if you are able to reproduce the bug on a maintained version (9.1.x and 9.2.x)?

Thanks!

**Comment by boo-code** [comment]
@paulnoelcholot Yes, it still reproduces on 9.2.x, with `ps_emailsubscription` 3.0.0 as bundled.

Measured by calling the module's own hooks with `NW_CONFIRMATION_EMAIL` enabled and the send methods instrumented, so the result does not depend on any mail transport:

```
on account CREATE with the newsletter box ticked : []
on the same subscription made later (0 -> 1)     : ["confirmation:probe-17097-later@example.com"]
```

So subscribing during registration sends nothing, while ticking the same box afterwards on an existing account sends the confirmation email. That asymmetry is the bug.

The reason is a `return` placed above the code that sends it. In `hookActionCustomerAccountAdd()`:

```php
if ($params['newCustomer']->newsletter) {
    return (bool) Db::getInstance()->execute('DELETE FROM ' . _DB_PREFIX_ . 'emailsubscription WHERE ...');
}
...
if ($newsletter) {
    if (Configuration::get('NW_CONFIRMATION_EMAIL')) {
        $this->sendConfirmationEmail($params['newCustomer']->email);
    }
```

The second block is only reachable when the customer did **not** subscribe, which is precisely when it must not run — so the confirmation email has never been sent from this path. On the module's `dev` branch that block no longer exists at all: PR PrestaShop/ps_emailsubscription#121 (merged 2026-08-29) removed it as dead code, which it was.

@Codencode's PR PrestaShop/ps_emailsubscription#139 is open and covers both the verification and the confirmation email for this path, so this issue looks like it needs review there rather than another fix.
